tests/pipeline/test_pipeline.py

```diff
@@ -235,6 +236,53 @@ def test_with_custom_config(self, snapshot: SnapshotTest):
 
         snapshot.assert_match(enriched_pipeline, "test-pipeline")
 
+    def test_with_custom_config_with_absolute_defaults_path(
+        self, snapshot: SnapshotTest
+    ):
+        with open(
+            str(RESOURCE_PATH / "custom-config/config.yaml"), "r"
+        ) as rel_config_yaml:
+            config_dict = yaml.safe_load(rel_config_yaml)
+        defaults_path = (
+            RESOURCE_PATH / "custom-config" / Path(config_dict["defaults_path"])
+        )
+        config_dict["defaults_path"] = str(defaults_path)
+        abs_config_yaml = open(
+            str(RESOURCE_PATH / "custom-config/temp_config.yaml"), "w"
+        )
+        yaml.dump(config_dict, abs_config_yaml)
+        result = runner.invoke(
+            app,
+            [
+                "generate",
+                "--pipeline-base-dir",
+                PIPELINE_BASE_DIR,
+                str(RESOURCE_PATH / "custom-config/pipeline.yaml"),
+                "--config",
+                str(RESOURCE_PATH / "custom-config/temp_config.yaml"),
```

Comment:
you're declaring this path a few times. Please create a variable for it, e.g. `abs_config_path`. Instead of `os.remove` you can then directly do `abs_config_path.unlink()`

Reply:
Good point, done

DIdn't know about `unlink()`, very nice! No need to import `os` :)